agent/client/workflow_listen_for_poll_requests.go

```diff
@@ -10,20 +10,27 @@ import (
 )
 
 func (c *Client) startPollerListener() error {
-	ctx := context.Background()
+	ctx, cancelCtx := context.WithCancel(context.Background())
 
 	client := proto.NewOrchestratorClient(c.conn)
 
 	request, err := c.getConnectionRequest()
 	if err != nil {
+		cancelCtx()
 		return err
 	}
 
 	stream, err := client.RegisterPollerAgent(ctx, request)
 	if err != nil {
+		cancelCtx()
 		return fmt.Errorf("could not open agent stream: %w", err)
 	}
 
+	go func() {
+		<-c.done
+		cancelCtx()
+	}()
```

Comment:
Initially, I was going to use the same goroutine for that, however, `stream.RecvMsg(&resp)` is a blocking operation. Meaning that if we got a shutdown command before receiving a polling request, it would just freeze because the server would probably remove this agent from the agent poll.

By using another goroutine, this doesn't happen. We can cancel the context as soon as we receive the shutdown message.

Reply:
Does it make sense to add a comment here too?

Reply:
managed to solve this issue in a better place, so no explanation is needed.